adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicPartitioners.scala

```diff
@@ -89,6 +89,16 @@ case class GenomicPositionPartitioner(numParts: Int, seqLengths: Map[String, Lon
         )
         getPart(refpos.referenceName, refpos.pos)
       }
+      case (refpos: ReferencePosition, k: Any) => {
+        require(
```

Comment:
I think having them as separate cases is OK. I might pull the `require(...)` into the `getPart`, but this doesn't "smell" too bad to me.